src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1119,6 +1120,8 @@ static ErrorCode getRefMismatchErrorCode(TypeKind type)
                 return ReportUnsafeIfNotAllowed(syntax, diagnostics);
             }
 
+            ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, selectedMethod, location);
```

Comment:
>ReportDiagnosticsIfUnmanagedCallersOnly [](start = 12, length = 39)

Which test(s) are covering delegate conversion scenario? 
Feels like the error message `ERR_UnmanagedCallersOnlyMethodsCannotBeCalledDirectly` may not be clear for such case.